packages/alsatian/core/running/callback-listener-informer.ts

```diff
@@ -0,0 +1,9 @@
+import {CallbackFunction} from "../events/generic-callback";
+
+export class ListenerInformer {
+	public async informListeners<B, T extends CallbackFunction<B>>(callbackFunctions: Array<T>, parameter: B) {
+		for (const callbackFunction of callbackFunctions) {
+			await callbackFunction(parameter);
```

Comment:
How does everything hold together if we don't `await`? I'm wondering in the case we have asynchronous handlers should we just let them execute in their own time as there is no benefit from our perspective to wait for the promise to resolve. Should we also consider `Error`s being thrown?

Reply:
If we do in fact want to await we should probably change this to
```suggestion
			return Promise.all(callbackFunctions.map(fn => fn(parameter));
```

This is probably better as it let's the caller decide whether to wait or not. The question above still applies to when this method is called though.

Reply:
The problem i had was that as i report the test execution via a rest interface i sometimes got the "test finish" call before "test start" call :)
If you use console.log you dont have this problem as it is always a synchronized call. I think you can still have an async function which does not use await so you are not forced to wait or im a wrong? I think if we do it like you wrote the problem with the order could still be the case i think...

Reply:
With respect to this it shouldn't be happening the important part I would expect is the `await` of the function calling this. All this is doing is getting all callback promises in one and wrapping them in a single promise so they can all be awaited further down the line which blocks other callbacks from happening but avoids having to wait for all callbacks for a single event to happen in serial.